src/ProductBundle/Entity/BaseProduct.php

```diff
@@ -243,9 +244,14 @@ public function getSku()
     /**
      * {@inheritdoc}
      */
-    public function setSlug($slug)
+    public function setSlug($slug, Slugify $slugifier = null)
```

Comment:
Better solutions imply events, like doctrine prePersist or preUpdate events, but not sure it will fit our use-case…